src/components/base-button/main.ts

```diff
@@ -0,0 +1,102 @@
+import { computed, reactive, PropType, SetupContext } from "vue";
+
+interface BaseButtonProps {
+  tag?: string;
+  type?: string;
+  size?: string;
+  textColor?: string;
+  nativeType?: string;
+  icon?: string;
+  text?: string;
+  outline?: boolean;
+  rounded?: boolean;
+  iconOnly?: boolean;
+  block?: boolean;
+}
+
+export default {
+  name: "BaseButton",
+  props: {
+    tag: {
+      type: String as PropType<BaseButtonProps>,
+      default: "button",
+      description: "Button tag (default -> button)",
+    },
+    type: {
```

Comment:
The way it is declared, would TS give me an error or warning if I try to specify a value which is not allowed?

Reply:
Its giving only warning